### ISSUE TITLE
Read only sessions: warn if a user has a smaller viewport

### ISSUE THREAD

**Issue by grahamc** [tmate-io/tmate#201]
I often want to share my terminal with colleagues or community members. I really like the ability to grant only read-only access. However, when I send a read-only session to someone, I have to communicate out of band with all of them to establish a viewport size which works for everyone. I understand why the read-only mode doesn't automatically resize for the smallest, but I would like to at least know if a viewer has a smaller port.

**Comment by nviennot** [comment]
It's because someone in the audience could potentially make the viewport too small (e.g. 2x2) for everyone, ruining the fun

I'm not sure what is an acceptable way to deal with the problem. Maybe a configuration flag

**Comment by grahamc** [comment]
Yep, understood.  I understand why the read-only mode doesn't automatically resize for the smallest. I'd at least like know if a viewer has a smaller port.

**Comment by heyakyra** [comment]
> It's because someone in the audience could potentially make the viewport too small (e.g. 2x2) for everyone, ruining the fun

Doesn't this already happen for non-readonly sessions? Why prevent it just for readonly?

**Comment by grahamc** [comment]
> Why prevent it just for readonly?

I might post a read-only view link to hundreds/thousands of anonymous users.

**Comment by heyakyra** [comment]
Ah I see…but in that case isn't it much harder to negotiate a size with that many people by hand?

**Comment by grahamc** [comment]
Sure. When sharing with many many people the signal is not useful. But sharing with 5-10, it would be reasonable to attempt to negotiate.
